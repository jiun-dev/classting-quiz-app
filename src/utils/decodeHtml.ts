export const decodeHTMLEntities = (text: string) => {
  return text
    .replace(/&quot;/g, '"')
    .replace(/&#039;/g, "'")
    .replace(/&amp;/g, '&');
};
