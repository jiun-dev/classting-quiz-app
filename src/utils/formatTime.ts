export const formatTime = (seconds: number): string => {
  const pad = (num: number) => num.toString().padStart(2, '0');
  const hours = pad(Math.floor(seconds / 3600));
  const minutes = pad(Math.floor((seconds % 3600) / 60));
  const secs = pad(seconds % 60);
  return `${hours}:${minutes}:${secs}`;
};
