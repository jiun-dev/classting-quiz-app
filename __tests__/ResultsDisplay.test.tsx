import ResultsDisplay from '../src/components/ResultDisplay';
import {render} from '@testing-library/react-native';
import {formatTime} from '../src/utils/formatTime';

describe('ResultsDisplay 컴포넌트 테스트', () => {
  test('컴포넌트 렌더링 테스트', () => {
    const totalQuestions = 10;
    const correctCount = 7;
    const elapsedTime = 1000;

    const {getByText} = render(
      <ResultsDisplay
        totalQuestions={totalQuestions}
        correctCount={correctCount}
        elapsedTime={elapsedTime}
      />,
    );

    expect(getByText(`총 퀴즈 개수: ${totalQuestions}`)).toBeDefined();
    expect(getByText(`정답: ${correctCount}`)).toBeDefined();
    expect(getByText(`오답: ${totalQuestions - correctCount}`)).toBeDefined();
    expect(getByText(`소요시간: ${formatTime(elapsedTime)}`)).toBeDefined();
  });
});

function getTempComponent(props: any): React.ReactElement {
  return <ResultsDisplay {...props} />;
}

let props;
let component: React.ReactElement;

describe('ResultsDisplay 컴포넌트 스냅샷 테스트', () => {
  props = {};
  component = getTempComponent(props);

  test('컴포넌트 렌더링 테스트', () => {
    const rendered = render(component);
    expect(rendered).toMatchSnapshot();
    expect(rendered).toBeTruthy();
  });
});
