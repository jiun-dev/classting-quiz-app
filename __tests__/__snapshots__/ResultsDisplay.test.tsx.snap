// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsDisplay 컴포넌트 스냅샷 테스트 컴포넌트 렌더링 테스트 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "400",
      }
    }
  >
    총 퀴즈 개수: 
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "400",
      }
    }
  >
    정답: 
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "400",
      }
    }
  >
    오답: 
    NaN
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "400",
      }
    }
  >
    소요시간: 
    NaN:NaN:NaN
  </Text>
</View>
`;

exports[`ResultsDisplay 컴포넌트 테스트 컴포넌트 렌더링 테스트 1`] = `
<View
  style={
    {
      "alignItems": "flex-start",
    }
  }
>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "400",
      }
    }
  >
    총 퀴즈 개수: 
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "400",
      }
    }
  >
    정답: 
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "400",
      }
    }
  >
    오답: 
    NaN
  </Text>
  <Text
    style={
      {
        "color": "#000000",
        "fontSize": 22,
        "fontWeight": "400",
      }
    }
  >
    소요시간: 
    NaN:NaN:NaN
  </Text>
</View>
`;
